docs/content/guides/dagster-pipes/subprocess/modify-external-code.mdx

```diff
@@ -0,0 +1,248 @@
+---
+title: "Using Dagster Pipes, Part two: Modify external code | Dagster Docs"
+description: "With Dagster Pipes, you can incorporate existing code into Dagster without huge refactors. This guide shows you how to modify existing code to work with Dagster Pipes."
+---
+
+# Modify external code to send information back to Dagster via Dagster Pipes
+
+<Note>
+
+This is part two of the [Dagster Pipes + subprocess](/guides/dagster-pipes/subprocess) tutorial.
+
+</Note>
+
+At this point, you should have two files:
+
+- `external_code.py` which is a standalone Python script that you want to orchestrate with Dagster.
+- `dagster_subprocess_pipes.py` which includes a Dagster asset and a Dagster definitions.
+
+In this section, you'll learn how to modify the standalone Python script to work with [Dagster Pipes](/guides/dagster-pipes) in order to stream information back to Dagster. To do this, you'll:
+
+- [Make Dagster context available in external code](#step-1-make-dagster-context-available-in-external-code)
+- [Stream log messages back to Dagster](#step-2-send-log-messages-to-dagster)
+- [Stream structured metadata back to Dagster](#step-3-send-structured-metadata-to-dagster)
+
+---
+
+## Step 1: Make Dagster context available in external code
+
+Getting external code to send information back to Dagster via Dagster Pipes requires adding a few lines of code:
+
+- Imports from `dagster-pipes`
+
+- A call that connects to Dagster Pipes: <PyObject module="dagster_pipes" object="open_dagster_pipes"/> initializes the Dagster Pipes context that can be used to stream information back to Dagster. It is recommended to call this function near the entry point of a pipes session.
+
+  <Note>
+
+  The <code>with open_dagster_pipes():</code> is a context manager in Python, ensuring resource setup and cleanup for a specific segment of code. It's useful for tasks requiring initial setup and final teardown, like opening and closing connections. In this case, the context manager is used to initialize and close the Dagster Pipes connection.
```

Comment:
i switched the order. i made it a callout because i thought it was an explanation to the python-specific knowledge "context manager" not pipes specific. the reason im explaining it is certain stakeholder audience may not be familiar with the usage of context manager.